pkg/scheduler/apis/config/types.go

```diff
@@ -270,3 +271,53 @@ func (p *Plugins) Append(src *Plugins) {
 	p.PostBind = appendPluginSet(p.PostBind, src.PostBind)
 	p.Unreserve = appendPluginSet(p.Unreserve, src.Unreserve)
 }
+
+// Apply merges custom plugins into default plugins.
+func (p *Plugins) Apply(customPlugins *Plugins) {
+	if customPlugins == nil {
+		return
+	}
+
+	p.QueueSort = mergePluginSets(p.QueueSort, customPlugins.QueueSort)
+	p.PreFilter = mergePluginSets(p.PreFilter, customPlugins.PreFilter)
+	p.Filter = mergePluginSets(p.Filter, customPlugins.Filter)
+	p.PostFilter = mergePluginSets(p.PostFilter, customPlugins.PostFilter)
+	p.Score = mergePluginSets(p.Score, customPlugins.Score)
+	p.Reserve = mergePluginSets(p.Reserve, customPlugins.Reserve)
+	p.Permit = mergePluginSets(p.Permit, customPlugins.Permit)
+	p.PreBind = mergePluginSets(p.PreBind, customPlugins.PreBind)
+	p.Bind = mergePluginSets(p.Bind, customPlugins.Bind)
+	p.PostBind = mergePluginSets(p.PostBind, customPlugins.PostBind)
+	p.Unreserve = mergePluginSets(p.Unreserve, customPlugins.Unreserve)
+}
+
+func mergePluginSets(defaultPluginSet, customPluginSet *PluginSet) *PluginSet {
```

Comment:
nit on style: you can avoid mentioning the type in the variable names, specially for unexported functions.  `default` and `custom` should be enough. But `default` is a reserved word, so `def` or `orig`.

Reply:
> nit on style: you can avoid mentioning the type in the variable names, specially for unexported functions. `default` and `custom` should be enough. 
> But `default` is a reserved word.

That is why I choose defaultPluginSet and customPluginSet here.

> so `def` or `orig`.

`def` and `orig` doesn't sounds like a good name to me. 

I'll keep this unless you feel strong about this.

Reply:
sg

Reply:
Thanks